app/static/js/scraper.js

```diff
@@ -2,22 +2,49 @@ let inputs = document.getElementsByTagName('textarea'),
     cacheCheckboxes = document.getElementsByClassName('cache'),
     submit = document.getElementById('submit');
 
-oninput = function() {
+function setSubmitEnabled() {
     let completed = true;
     for (let input of inputs) {
-        if (!Boolean(input.value)) {
+        if (!Boolean(input.value.trim())) {
             completed = false;
             break;
         }
     }
     submit.disabled = !completed;
 }
 
+function saveInputsToURL() {
+    const params = new URLSearchParams();
+    for (let input of inputs) {
+        params.append(input.name, encodeURIComponent(input.value));
+    }
+    this.history.pushState({}, '', '?' + params.toString());
```

Comment:
Very clever! Separately, we should do this for the search on the homepage too... #16